src/components/MainNav/ProfileDropdown/ProfileDropdown.js

```diff
@@ -89,8 +89,15 @@ class ProfileDropdown extends Component {
 
   createLink(link, index, module) {
     const { stripes } = this.props;
-    const { check } = link;
-    const checkfn = !check ? undefined : (module.getModule()[check] || validations[check]);
+    const { check, route } = link;
+    const checkingFoo = module.getModule()[check] || validations[check];
+    let checkfn;
+
+    if (route === '/settings/myprofile/password') {
+      checkfn = !check ? undefined : (stripes.hasPerm('ui-myprofile.settings.change-password') && checkingFoo);
```

Comment:
Please try to avoid negative if statement like this. Use check ? (stripes.hasPerm....) : null; Also it is a bad practice assign undefined, I would suggest use null in this cases.